src/flask/app.py

```diff
@@ -1167,17 +1168,20 @@ def make_response(self, rv: ft.ResponseReturnValue) -> Response:
 
         # unpack tuple returns
         if isinstance(rv, tuple):
-            len_rv = len(rv)
-
             # a 3-tuple is unpacked directly
-            if len_rv == 3:
-                rv, status, headers = rv  # type: ignore[misc]
+            if len(rv) == 3:
```

Comment:
I don't understand why this code was refactored. Can you address the issue with more minimal code changes?

Reply:
Please report that issue to Pyright and link it here so we can track when that's fixed. I don't like the idea of refactoring this in this way just to satisfy pyright. This is code that runs for _every single response_ and so adding extra calls is not free.

Reply:
Prefer `ignore`, since a comment has no runtime cost.